src/Slack/UserExtractor.php

```diff
@@ -11,37 +11,39 @@
 
 namespace JoliCode\SecretSanta\Slack;
 
-use CL\Slack\Model\User as SlackUser;
-use CL\Slack\Payload\UsersListPayload;
-use CL\Slack\Payload\UsersListPayloadResponse;
 use JoliCode\SecretSanta\Exception\UserExtractionFailedException;
 use JoliCode\SecretSanta\User;
+use JoliCode\Slack\Api\Model\ObjsUser;
+use JoliCode\Slack\ClientFactory;
 
 class UserExtractor
 {
-    private $apiHelper;
-
-    public function __construct(ApiHelper $apiHelper)
-    {
-        $this->apiHelper = $apiHelper;
-    }
-
     public function extractAll(string $token): array
     {
-        $payload = new UsersListPayload();
+        $client = ClientFactory::create($token);
 
-        try {
-            /** @var $response UsersListPayloadResponse */
-            $response = $this->apiHelper->sendPayload($payload, $token);
-        } catch (\Throwable $t) {
-            throw new UserExtractionFailedException('Could not fetch members in team', 0, $t);
-        }
+        /** @var ObjsUser[] $slackUsers */
+        $slackUsers = [];
+        $cursor = '';
+
+        do {
+            try {
+                $response = $client->usersList([
+                    'limit' => 200,
+                    'cursor' => $cursor,
+                ]);
+            } catch (\Throwable $t) {
+                throw new UserExtractionFailedException('Could not fetch members in team', 0, $t);
+            }
+
+            $slackUsers = array_merge($slackUsers, $response->getMembers());
+            $cursor = $response->getResponseMetadata() ? $response->getResponseMetadata()->getNextCursor() : '';
+        } while (!empty($cursor));
```

Comment:
This could go into a request timeout from the server, if we loop a lot. We should add a limit like when we send the messages to avoid server error. In that case, it's OK to display a message to the user like "your team is too big".